src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,651 @@
+		}
+
+		public String[][] getUrlEncTimedValueInjection() {
+			return urlEncTimedValueInjection;
+		}
+		
+		public String getPrefix() {
+			return prefix;
+		}
+		
+	};
+
+   private static final Logger log = Logger.getLogger(TestNoSQLInjection.class);
+
+	@Override
+	public int getId() {
+		return ID;
+	}
+
+	
+	 @Override
+	 public boolean targets(TechSet technologies) {
+		if (technologies.includes(Tech.Db)) {
+			return true;
+		}
+
+		for (Tech tech : technologies.getIncludeTech()) {
+			if (tech.getParent() == Tech.Db) {
+				return true;
+			}
+		}
+		return false;	
+	 }
+	
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX+ "name");
+	}
+
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+   
+	@Override
+	public int getCweId() {
+		return CWED_ID;
+	}
+
+	@Override
+	public int getWascId() {
+		return WASC_ID;
+	}
+	
+	
+    /**
+     * Scan for NoSQL injection vulnerabilities
+     * @inheritDoc
+     *
+     */
+	public void scan(HttpMessage msg, String param, String value) {
+
+		String paramInjected, valueInjected, attackType;
+		HttpMessage injectedMsg, baseMsg;
+		long intervalBaseMessage, intervalInjectedMessage; 
+		Instant now;
+		boolean onlyExactMatch;
+
+		try {
+			
+			baseMsg = getNewMsg();
+			setParameter(baseMsg, param, TOKEN);
+
+			now = Instant.now();
+			sendAndReceive(baseMsg);
+			intervalBaseMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+			
+			
+			for(NOSQLDB nosql: NOSQLDB.values()) {
+
+				if(inScope(nosql.getDbTech())){
+
+					
+					/**
+					 * TODO check for the URL_ENCEODED format
+					 * 
+					 */
+					if (true) { 
+
+						for(String[] pv : nosql.getUrlEncParamValueInjection()) {
+							
+							onlyExactMatch=false;
+							attackType = PARVAL_ATTACK;
+							injectedMsg = getNewMsg();
+							paramInjected = param+pv[0];
+							valueInjected = pv[1];
+							
+							setParameter(injectedMsg, paramInjected, valueInjected);
+		
+							try {
+								sendAndReceive(injectedMsg, false);
+
+								if(isBingo(baseMsg, injectedMsg, paramInjected, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+		
+							} catch (SocketException ex) {
+		
+								if (log.isDebugEnabled()) 
+									log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+											" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;

```

Comment:
Should use curly braces to be consistent with the rest of the conditionals. (Throughout)

Same for debug statements, etc.